src/cargo/core/resolver/mod.rs

```diff
@@ -487,6 +527,7 @@ fn activate_deps_loop(
                                 &parent,
                                 backtracked,
                                 &conflicting_activations,
+                                check_public_visible_dependencies,
```

Comment:
Could this perhaps be stored in a `Context` or somewhere to avoid being passed around?

Reply:
It could, and if you want to do it will be. Generally, Context storage things that change as resolution precedes. This will always have the same value. So I didn't see the need to store it repeatedly on the heap. Of course it's one bool, so it doesn't matter very much.

Would you like me to change it?

Reply:
Yeah it's not the greatest spot for it, but I figure it's a bit better than passing and threading around to lots of places

Reply:
Does that look better?